### PR TITLE
Generate a getter method for casttype fields

### LINKED ISSUE #87: Bug: casttype disables getter methods
Adding a `casttype` to a field removes its getter method. Is this intended? Doesn't seem so.

### PR DESCRIPTION
**Pull request by tamird**
Closes #87.

@awalterschulze can you add enable travis for this repo? (alternatively, I can do it for you if you make me an admin)

### CODE REVIEW

**Comment by awalterschulze** [comment]
Why always pushing the travis? :)
If you want matrix builds, what about different protoc versions?

The fix seems solid, I just think I would like another one of these
https://github.com/gogo/protobuf/blob/master/test/casttype/casttype.proto
Where 
```
option (gogoproto.goproto_getters_all) = true;
option (gogoproto.face_all) = false;
```

**Comment by tamird** [comment]
> Why always pushing the travis? :)

Because every time I make a PR to this repo I have to fix existing tests.

> If you want matrix builds, what about different protoc versions?

Done.

> The fix seems solid, I just think I would like another one of these

Added.

**Comment by tamird** [comment]
Working on getting Travis to actually pass. Will ping when it does.

**Comment by awalterschulze** [comment]
I prefer drone.io, but if you can get matrix builds with protoc and go versions going in travis I will have to change my mind.  Looking at your diff seems like you are well on your way to convincing me.

**Comment by tamird** [comment]
@awalterschulze ok this is good to merge. here's the passing travis build from my fork: https://travis-ci.org/tamird/protobuf-1/builds/74555110

File: vanity/test/Makefile [hunk 1]
```diff
@@ -11,8 +11,8 @@ regenerate:
 	protoc --gogoslick_out=./slick/ vanity.proto
 	protoc --proto_path=../../:../../../../../:../../protobuf/:. --gogoslick_out=./slick/ gogovanity.proto
 	protoc-min-version -version="3.0.0" --proto_path=../../:../../../../../:../../protobuf/:. --gogoslick_out=./slick/ proto3.proto
-	
+
 test:
 	go install github.com/gogo/protobuf/protoc-gen-gofast
 	protoc --gofast_out=./gofast/ vanity.proto
-	go test -v ./...
+	go test ./...
```
Comment (awalterschulze) [inline vanity/test/Makefile]: Why not -v?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+env:
+  - PROTOBUF_VERSION=2.6.1
```
Comment (awalterschulze) [inline .travis.yml]: Can we have protobuf_version 2.3, 2.4.1, 2.5 as well?
Or at least 2.5, since that is what we are still using.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+install:
+  - go get -t ./...
+
+script:
+  - PATH=/home/travis/bin:$PATH make all
+
+language: go
+
+go:
+  - 1.3
```
Comment (awalterschulze) [inline .travis.yml]: What about 1.2.2?

**Comment by awalterschulze** [comment]
The matrix builds look really cool :)

File: Makefile [hunk 4]
```diff
@@ -90,42 +90,8 @@ regenerate:
...
-	go test -v ./test/custombytesnonstruct
-	go test -v ./test/required
-	go test -v ./test/casttype/...
-	go test -v ./vanity/test
-	go test -v ./test/sizeunderscore
-	go test -v ./test/issue34
-	go test -v ./test/empty-issue70
-	go test -v ./test/indeximport-issue72
-	make vet
+	go test ./...
```
Comment (awalterschulze) [inline Makefile]: Why not -v?

**Comment by awalterschulze** [comment]
How long do all the tests take to run?

**Comment by awalterschulze** [comment]
Is this the free part of travis for open source projects?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+env:
+  - PROTOBUF_VERSION=2.6.1
+  - PROTOBUF_VERSION=3.0.0-alpha-3
+
+before_install:
+  - ./install-protobuf.sh
+  - PATH=/home/travis/bin:$PATH protoc --version
+
+install:
+  - go get -t ./...
```
Comment (awalterschulze) [inline .travis.yml]: Why is this needed?
Gogoprotobuf should not have any dependencies.

File: Makefile (hunk 4 above)
Comment (tamird) [inline Makefile]: it's just spammy. if a test fails, you'll get the output whether you passed `-v` or not.

File: vanity/test/Makefile (hunk 1 above)
Comment (tamird) [inline vanity/test/Makefile]: same reason as above

File: .travis.yml (hunk 2 above)
Comment (tamird) [inline .travis.yml]: Added 2.5.0.

File: .travis.yml (hunk 5 above)
Comment (tamird) [inline .travis.yml]: removed.

File: .travis.yml (hunk 3 above)
Comment (tamird) [inline .travis.yml]: added 1.2

**Comment by tamird** [comment]
@awalterschulze now that I've added protobuf 2.5.0 and go 1.2 and 1.5, our matrix is 18 builds. It's going to take about 30-40 minutes to get through all of them, and they each take about 10 minutes.

**Comment by tamird** [comment]
Yes, this is free.

**Comment by tamird** [comment]
BTW, let me know when you're ready to merge this and I'll remove that last commit (it's necessary only on my fork, not in this repo).

**Comment by tamird** [comment]
Turns out travis doesn't know about go 1.5 yet, so I removed that bit.

**Comment by tamird** [comment]
OK, removed the last commit and here's the passing build from my fork: https://travis-ci.org/tamird/protobuf-1/builds/74585173

Each build took 4-6 minutes and total time was about 13m.

**Comment by awalterschulze** [comment]
Wow thats amazing :)

**Comment by tamird** [comment]
Don't forget to turn CI in on the Travis web interface :)

**Comment by awalterschulze** [comment]
Busy with that now

**Comment by awalterschulze** [comment]
Ok cool all seems to working.
Thank you very much, this is great :)

**Comment by awalterschulze** [comment]
The travis matrix builds are really cool, but they don't always work and when I want to view the log they take forever to load.  Granted I am on a slow line, but I did not have these problems with drone.io

**Comment by awalterschulze** [comment]
Just btw
https://github.com/gogo/protobuf/issues/93
